pages/accountLists/[accountListId].page.test.tsx

```diff
@@ -108,4 +108,91 @@ describe('AccountListsId page', () => {
       expect(getByText('Good Morning, firstName.')).toBeInTheDocument();
     });
   });
+  describe('AccountListId Redirects', () => {
+    const restApiNotFoundErrorMessage = "Resource 'AccountList' is not valid";
+
+    it('replaces and redirects to the default account list id if Not Found', async () => {
+      const query = jest.fn();
+
+      (makeSsrClient as jest.Mock).mockReturnValue({
+        query: query,
+      });
+      query.mockRejectedValueOnce(new Error(restApiNotFoundErrorMessage));
+
+      query.mockResolvedValueOnce({
+        data: {
+          user: {
+            defaultAccountList: 'default-id',
+          },
+        },
+      });
+
+      const { redirect } = (await getServerSideProps({
+        req: { url: '/accountLists/[accountListId]/contacts' },
+        query: {
+          accountListId: 'account-list-1',
+        },
```

Comment:
You could make a `makeContext` helper function that takes a URL and returns this context object. Then you could use `makeContext` in each test to reduce some of the duplication.